ui_framework/components/button/button.js

```diff
@@ -0,0 +1,165 @@
+import React, {
+  PropTypes,
+} from 'react';
+
+import classNames from 'classnames';
+import keyMirror from 'keymirror';
+
+import { KuiButtonIcon } from './button_icon/button_icon';
+
+const commonPropTypes = {
+  type: PropTypes.oneOf([
+    'basic',
+    'hollow',
+    'danger',
+    'primary',
+  ]),
+  testSubject: PropTypes.string,
+  isDisabled: PropTypes.bool,
+  onClick: PropTypes.func,
+  data: PropTypes.any,
+  className: PropTypes.string,
+};
+
+const nonVoidPropTypes = {
```

Comment:
I don't understanding the naming here.  What makes these `nonVoid` and different from the props in `common`?